File: horizonBackend/apps/cinema/views/cinema.py
from rest_framework.generics import ListAPIView
from rest_framework.views import APIView
from apps.cinema.models import Cinema
from apps.cinema.serializers.cinema import CinemaSerializer
from utils.response import RetResponse
from utils.pagination import CustomPageNumberPagination

class CinemaView(APIView):
  authentication_classes = []
  def post(self,request):
    name = request.data.get("name")
    alias = request.data.get("alias")
    address = request.data.get("address")
    userId = request.data.get("userId")
    print(userId)
    Cinema.objects.create(
      name=name,
      alias = alias,
      address = address,
      user_id = userId
    )
    return RetResponse.success(None,None)
class CinemaUserView(ListAPIView):
  authentication_classes = []
  serializer_class = CinemaSerializer
  pagination_class = CustomPageNumberPagination
  def get_queryset(self):
    user_id = self.kwargs.get("id")
    queryset = Cinema.objects.filter(user_id=user_id)
    return queryset

File: horizon-manage/src/api/modules/cinema.js
import http from "@/api";
export function getUserCinemaApi(data) {
  return http.get(`/cinema/user/${data.userId}`, data);
}
export function createCinemaApi(data) {
  return http.post("/cinema/create", data);
}
export function getCinemaTagApi(data) {
  return http.get("/cinema/tag/list", data);
}
export function setCinemaTagApi(data) {
  return http.post("/cinema/tag", data);
}
